conn.go

```diff
@@ -1,17 +1,24 @@
 package pool
 
-import "net"
+import (
+	"net"
+	"sync"
+)
 
 // PoolConn is a wrapper around net.Conn to modify the the behavior of
 // net.Conn's Close() method.
 type PoolConn struct {
 	net.Conn
+	sync.RWMutex
```

Comment:
This will expose the lock outside the package. Can you assign it to an unexported field, such as `mu sync.RWMutex`, would prefer that